src/lib/modernExtend.ts

```diff
@@ -362,780 +751,1213 @@ export function light(args?: LightArgs): ModernExtend {
-    const exposes = [e.lock(), e.pincode(), e.lock_action(), e.lock_action_source_name(), e.lock_action_user(),
-        e.auto_relock_time().withValueMin(0).withValueMax(3600), e.sound_volume()];
-    const configure: Configure = async (device, coordinatorEndpoint, logger) => {
-        await setupAttributes(device, coordinatorEndpoint, 'closuresDoorLock', [
-            {attribute: 'lockState', min: 'MIN', max: '1_HOUR', change: 0}], logger);
-    };
-    const meta: DefinitionMeta = {pinCodeCount: args.pinCodeCount};
-
-    return {fromZigbee, toZigbee, exposes, configure, meta, isModernExtend: true};
+export interface CommandsLevelCtrl {
+    commands?: (
+        'brightness_move_to_level' | 'brightness_move_up' | 'brightness_move_down' | 'brightness_step_up' | 'brightness_step_down' | 'brightness_stop'
+    )[],
+    bind?: boolean, endpointNames?: string[]
 }
+export function commandsLevelCtrl(args?: CommandsLevelCtrl): ModernExtend {
+    args = {commands: [
+        'brightness_move_to_level', 'brightness_move_up', 'brightness_move_down', 'brightness_step_up', 'brightness_step_down', 'brightness_stop',
+    ], bind: true, ...args};
+    let actions: string[] = args.commands;
+    if (args.endpointNames) {
+        actions = args.commands.map((c) => args.endpointNames.map((e) => `${c}_${e}`)).flat();
+    }
+    const exposes: Expose[] = [
+        e.enum('action', ea.STATE, actions).withDescription('Triggered action (e.g. a button click)').withCategory('diagnostic'),
+    ];
 
-export interface EnumLookupArgs {
-    name: string, lookup: KeyValue, cluster: string | number, attribute: string | {ID: number, type: number}, description: string,
-    zigbeeCommandOptions?: {manufacturerCode?: number, disableDefaultResponse?: boolean}, access?: 'STATE' | 'STATE_GET' | 'ALL',
-    endpointName?: string, reporting?: ReportingConfigWithoutAttribute, entityCategory?: 'config' | 'diagnostic',
-}
-export function enumLookup(args: EnumLookupArgs): ModernExtend {
-    const {name, lookup, cluster, attribute, description, zigbeeCommandOptions, endpointName, reporting, entityCategory} = args;
-    const attributeKey = isString(attribute) ? attribute : attribute.ID;
-    const access = ea[args.access ?? 'ALL'];
+    const defaultSimulatedBrightness = 255;
+
+    const simulatedBrightness = [e.composite('simulated_brightness', 'simulated_brightness', ea.SET)
+        .withDescription(`Simulate a brightness value.` +
+        `If this device provides a brightness_move_up or brightness_move_down action it is possible to specify the update interval and delta.` +
+        `The action_brightness_delta indicates the delta for each interval.`)
+        .withFeature(e.numeric('delta', ea.SET).withValueMin(0).withDescription('Delta per interval, 20 by default'))
+        .withFeature(e.numeric('interval', ea.SET).withValueMin(0).withUnit('ms').withDescription('Interval duration'))];
+
+    const fromZigbee: Fz.Converter[] = [
```

Comment:
It seems we duplicate a lot of code from existing converters here, can't we just use e.g. `const fromZigbee = [fz.command_stop, fz.command_step, ...` here?

Reply:
I was thinking of removing those separate generic fz and tz after converting everything to modern extends to concentrate everything inside extends. What is the general strategy on keeping/removing separate converters? I see a lot of new devices being added that use those instead of existing modern extends

Reply:
I propose to move them only once no devices uses them directly. So for now just refer to them to prevent code duplication.

Reply:
Got it, will re-use them for now